lisa/transformers/upgrade_packages.py

```diff
@@ -0,0 +1,170 @@
+    ...
+
+
+@dataclass_json
+@dataclass
+class UpgradeInstallerTransformerSchema(schema.Transformer):
+    # SSH connection information to the node
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    # installer's parameters.
+    installer: Optional[BaseInstallerSchema] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+
+
+class BaseInstaller(subclasses.BaseClassWithRunbookMixin):
+    def __init__(
+        self,
+        runbook: Any,
+        node: Node,
+        parent_log: Logger,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(runbook, *args, **kwargs)
+        self._node = node
+        self._log = get_logger("upgrade_installer", parent=parent_log)
+
+    def validate(self) -> None:
+        raise NotImplementedError()
+
+    def install(self) -> str:
+        raise NotImplementedError()
+
+
+class UpgradeInstallerTransformer(Transformer):
+    @classmethod
+    def type_name(cls) -> str:
+        return "upgrade_installer"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return UpgradeInstallerTransformerSchema
+
+    @property
+    def _output_names(self) -> List[str]:
+        return []
+
+    def _internal_run(self) -> Dict[str, Any]:
+        runbook: UpgradeInstallerTransformerSchema = self.runbook
+        assert runbook.connection, "connection must be defined."
+        assert runbook.installer, "installer must be defined."
+
+        node = quick_connect(runbook.connection, "installer_node")
+
+        uname = node.tools[Uname]
+        self._log.info(
+            f"kernel version before install: {uname.get_linux_information()}"
+        )
+        factory = subclasses.Factory[BaseInstaller](BaseInstaller)
+        installer = factory.create_by_runbook(
+            runbook=runbook.installer, node=node, parent_log=self._log
+        )
+
+        installer.validate()
+        package_list = installer.install()
+        self._log.info(f"Packages upgraded: {package_list}")
+
+        self._log.info("rebooting")
+        node.reboot()
+        self._log.info(
+            f"kernel version after install: "
+            f"{uname.get_linux_information(force_run=True)}"
+        )
+
+        return {}
+
+
+class UnattendedUpgradeInstaller(BaseInstaller):
+    def __init__(
+        self,
+        runbook: Any,
+        node: Node,
+        parent_log: Logger,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(runbook, node, parent_log, *args, **kwargs)
+        self.repo_url = "http://archive.ubuntu.com/ubuntu/"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "unattended_upgrade"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BaseInstallerSchema
+
+    def validate(self) -> None:
+        assert isinstance(self._node.os, Debian), (
```

Comment:
How about make sure it's available for all distros? But so far, we just implement for Debian.